packages/nimble-components/src/table-column/number-text/models/unit-scale-formatter.ts

```diff
@@ -0,0 +1,91 @@
+import type { ScaledUnit } from './scaled-unit';
+import { NumberFormatter } from './number-formatter';
+
+export type UnitScaleFormatterConstructor = new (
+    locale: string,
+    formatterOptions: Intl.NumberFormatOptions
+) => UnitScaleFormatter;
+
+/**
+ * A formatter for a number-text column that supports unit labels.
+ */
+export abstract class UnitScaleFormatter extends NumberFormatter {
+    public alwaysUseBaseScaledUnit = false;
+    private _supportedScaledUnits?: ScaledUnit[];
+    private _baseScaledUnit?: ScaledUnit;
+
+    private get supportedScaledUnits(): ScaledUnit[] {
+        if (this._supportedScaledUnits === undefined) {
+            this.setSupportedScaledUnitsAndBaseScaledUnit();
+        }
+        return this._supportedScaledUnits!;
+    }
+
+    private get baseScaledUnit(): ScaledUnit {
+        if (this._baseScaledUnit === undefined) {
+            this.setSupportedScaledUnitsAndBaseScaledUnit();
+        }
+        return this._baseScaledUnit!;
+    }
+
+    public constructor(
+        private readonly locale: string,
+        private readonly formatterOptions: Intl.NumberFormatOptions
+    ) {
+        super();
+    }
+
+    public getScaledNumber(number: number): number {
```

Comment:
Just out of curiosity, what is the reasoning behind making this a public method? If it's just for tests, could you mark it `internal`?

Reply:
No, it is called from `DecimalFormatter` and `DefaultFormatter`. Those formatters make decisions based on what the ultimate displayed value will be, so they need to ask the `UnitScaleFormatter` for it.